deepinterpolation/inferrence_collection.py

```diff
@@ -233,45 +313,164 @@ def run(self):
+
+            start = first_sample + index_dataset * self.batch_size
+            end = first_sample + index_dataset * self.batch_size \
+                + local_size
+    
+            with h5py.File(self.output_file, "a") as file_handle:
+                dset_out = file_handle[output_dataset_name]
+                if self.save_raw:
+                    raw_out = file_handle[raw_dataset_name]
+                    if self.rescale:
+                        corrected_raw = local_data[1] * local_std + local_mean
+                    else:
+                        corrected_raw = local_data[1]
+
+                    raw_out[start:end] = np.squeeze(corrected_raw, -1)
+
+                # We squeeze to remove the feature dimension from tensorflow
+                dset_out[start:end] = np.squeeze(corrected_data, -1)
+
+    def run_multiprocessing(self):
+
+        if self.output_padding:
+            first_sample = self.generator_obj.start_sample - \
+                self.generator_obj.start_frame
+            final_shape = [self.nb_datasets*self.batch_size+first_sample]
 
-            for epoch_index, index_dataset in enumerate(tqdm(np.arange(0, self.nb_datasets, self.steps_per_epoch))):
+        else:
+            final_shape = [self.nb_datasets*self.batch_size]
+            first_sample = 0
 
-                local_length = np.min([self.steps_per_epoch, self.nb_datasets-index_dataset])
-                
-                # We overwrite epoch_index to allow the last unfilled epoch
-                self.generator_obj.epoch_index = epoch_index
+        final_shape.extend(self.indiv_shape[:-1])
 
-                predictions_data = self.model.predict(
-                        self.generator_obj,
-                        steps = local_length, max_queue_size = 10,
-                        workers = self.workers,
-                        use_multiprocessing = self.use_multiprocessing)
+        chunk_size = [1]
+        chunk_size.extend(self.indiv_shape[:-1])
 
-                local_mean, local_std = \
-                        self.generator_obj.__get_norm_parameters__(index_dataset)                
-                
-                local_size = predictions_data.shape[0]
+        output_dataset_name = "data"
+        raw_dataset_name = "raw"
 
-                if self.rescale:
-                    corrected_data = predictions_data * local_std + local_mean
-                else:
-                    corrected_data = predictions_data
-                
-                start = local_start
-                end = local_start + local_size - 1
+        with h5py.File(self.output_file, "w") as file_handle:
+            file_handle.create_dataset(
+                output_dataset_name,
+                shape=tuple(final_shape),
+                chunks=tuple(chunk_size),
+                dtype=self.output_datatype,
+            )
 
-                # We adjust for the next for loop run
-                local_start = end + 1
+            if self.save_raw:
```

Comment:
this is duplicated in both `run` and here. Please abstract out duplicated parts so there is no more duplication